CONTRIBUTING.md

```diff
@@ -1,14 +1,27 @@
 # Developing the code
 
-An appropriate environment for developing a package can be entered using `nix-shell` in the package directory. This
-includes all the dependencies for the package and `cabal`, so you should be able to `cabal build` a package after
-doing this.
+You can get an environment for developing the entire project using `nix-shell` in the root directory.
+This will have all external dependencies present, so is suitable for building the packages with e.g. `cabal v2-build`.
+
+An environment for developing a particular package in isolation can be entered by using `nix-shell` in the
+package directory. This has all the dependencies, including local ones, provided. This shouldn't be necessary
+any more, but is useful if you need to use an old-style `cabal` command that only works in a single package context.
+
+You can also use `cabal` and `stack` outside a Nix environment to build the project. *However* there are two caveats:
+- You may get different versions of packages.
+    - This is more of a problem for `cabal` than `stack`, since our Nix package set is based off 
+      the Stackage resolver.
```

Comment:
We can use [`--pvp-bounds`](https://www.fpcomplete.com/blog/2015/09/stack-pvp) to mitigate the issue.

Reply:
Or we should use a freeze file, but those don't seem terribly well-supported so far.